src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/VaultContentXMLContentCreator.java

```diff
@@ -0,0 +1,366 @@
+    }
+    
+
+    @Override
+    public boolean switchCurrentNode(String subPath, String newNodeType) throws RepositoryException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void createUser(String name, String password, Map<String, Object> extraProperties) throws RepositoryException {
+        try {
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB  .append("create user " + name + " with password " + password + " \n\n")
+                            .append("set properties on authorizable(" + name + ")\n");
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createGroup(String name, String[] members, Map<String, Object> extraProperties) throws RepositoryException {
+        try {
+            StringBuilder repoInitTextSB = new StringBuilder();
+            repoInitTextSB  .append("create group " + name + " \n\n")
+                            .append("add " + String.join(",", members) + " to group " + name)
+                            .append("set properties on authorizable(" + name + ")\n");
+            extraProperties.entrySet().stream().map(this::getSetPropertyString).forEach(repoInitTextSB::append);
+            repoInitTextSB.append("end");
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createAce(String principal, String[] grantedPrivileges, String[] deniedPrivileges, String order) throws RepositoryException {
+        try {
+            //set principal ACL for principal1,principal2
+            //allow jcr:read
+            StringBuilder repoInitTextSB = new StringBuilder();
+            String path = this.currentNode.getPath();
+            
+            repoInitTextSB.append("set ACL for " + principal +"\n\n");
+         
+            for(String privilege: grantedPrivileges){
+                repoInitTextSB.append("allow " + privilege + " on " + path + "\n");
+            }
+            for(String privilege: deniedPrivileges){
+                repoInitTextSB.append("deny " + privilege + " on " + path + "\n");
+            }
+            repoInitTextSB.append("end");
+            
+            repoInitTextExtensionConsumer.accept(repoInitTextSB.toString());
+        } catch (IOException | ConverterException e) {
+            throw new RepositoryException(e);
+        }
+    }
+
+    @Override
+    public void createAce(String principalId, String[] grantedPrivilegeNames, String[] deniedPrivilegeNames,
```

Comment:
- a principal only has a name and never an ID. please use 'principalName' or 'principal' (but keep it consistent with the other 'createAce' method
- order: same as above
- not sure i understand how this method is expected to work.... restrictions are always associated with a given single entry..... it doesn't make sense to pass separate maps of single/mv-restrictions and have an array of granted/denied privs..... it should be 'privileges', boolean isAllow plus the restrictions.